colossalai/utils/memory_tracer/memstats_collector.py

```diff
@@ -23,45 +23,71 @@ def reset(self):
 
 
 class MemStatsCollector:
+    """
+    A Memory statistic collector.
+    It works in two phases. 
+    Phase 1. Collection Phase: collect memory usage statistics of CPU and GPU.
+    The first iteration of DNN training.
+    Phase 2. Runtime Phase: use the read-only collected stats
+    The rest iterations of DNN training.
+    
+    It has a Sampling counter which is reset after DNN training iteration.
+    """
 
     def __init__(self) -> None:
-        """
-        Collecting Memory Statistics.
-        It has two phases. 
-        1. Collection Phase: collect memory usage statistics
-        2. Runtime Phase: do not collect statistics.
-        """
         self._sampling_cnter = SamplingCounter()
-        self._model_data_cuda = []
-        self._overall_cuda = []
+        self._model_data_cuda_list = []
+        self._overall_cuda_list = []
 
-        # TODO(jiaruifang) Now no cpu mem stats collecting
-        self._model_data_cpu = []
-        self._overall_cpu = []
+        self._model_data_cpu_list = []
+        self._overall_cpu_list = []
 
         self._start_flag = False
 
-    @property
-    def overall_cuda(self):
-        return self._overall_cuda
-
-    @property
-    def model_data_cuda_GB(self):
-        return [elem / 1e9 for elem in self._model_data_cuda]
-
-    @property
-    def model_data_cuda(self):
-        return self._model_data_cuda
-
-    @property
-    def non_model_data_cuda_GB(self):
-        return [elem / 1e9 for elem in self.non_model_data_cuda]
+    def overall_mem_stats(self, type: str):
+        if type == 'cuda':
+            return self._overall_cuda_list
+        elif type == 'cpu':
+            return self._overall_cpu_list
+        else:
+            raise TypeError
 
     @property
-    def non_model_data_cuda(self):
+    def model_data_cuda_list(self, type: str, unit: str = 'B') -> List[int]:
```

Comment:
maybe a function can reduce the duplicated codes?

Reply:
not a big deal. we can polish it later.

Reply:
DONE